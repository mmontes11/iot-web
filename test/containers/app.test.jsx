import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import App from "containers/app";

describe("components/app", () => {
  const state = {
    auth: {
      isAuth: false,
      username: null,
      password: null
    },
    request: {
      pending: 0,
      statusCode: null,
      error: null
    }
  };

  it("renders the app", () => {
    const store = configureStore([])(state);
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <App store={store} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
