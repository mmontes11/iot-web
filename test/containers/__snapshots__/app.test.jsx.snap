// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/app renders the app with auth 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/things",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/things",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Connect(App))
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Route>
        <Connect(App)
          history={
            Object {
              "action": "REPLACE",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/things",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/things",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/things",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": false,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <App
            dispatch={[Function]}
            history={
              Object {
                "action": "REPLACE",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/things",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/things",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isAuth={true}
            location={
              Object {
                "hash": "",
                "pathname": "/things",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            store={
              Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              }
            }
            storeSubscription={
              Subscription {
                "listeners": Object {
                  "clear": [Function],
                  "get": [Function],
                  "notify": [Function],
                  "subscribe": [Function],
                },
                "onStateChange": [Function],
                "parentSub": undefined,
                "store": Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                },
                "unsubscribe": [Function],
              }
            }
          >
            <Provider
              store={
                Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
              }
            >
              <Main>
                <div
                  className="has-navbar-fixed-top"
                >
                  <withRouter(Connect(Navbar))>
                    <Route>
                      <Connect(Navbar)
                        history={
                          Object {
                            "action": "REPLACE",
                            "block": [Function],
                            "canGo": [Function],
                            "createHref": [Function],
                            "entries": Array [
                              Object {
                                "hash": "",
                                "pathname": "/things",
                                "search": "",
                                "state": undefined,
                              },
                            ],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "index": 0,
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/things",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/things",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": false,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                      >
                        <Navbar
                          history={
                            Object {
                              "action": "REPLACE",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/things",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/things",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          isHamburgerMenuExpanded={false}
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/things",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          logout={[Function]}
                          match={
                            Object {
                              "isExact": false,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                          toggleHamburgerMenu={[Function]}
                        >
                          <nav
                            className="navbar has-shadow is-spaced"
                          >
                            <div
                              className="navbar-brand"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="navbar-item"
                                to="/"
                              >
                                <Route
                                  path="\\\\/"
                                >
                                  <Link
                                    aria-current="page"
                                    className="navbar-item active"
                                    replace={false}
                                    style={Object {}}
                                    to="/"
                                  >
                                    <a
                                      aria-current="page"
                                      className="navbar-item active"
                                      href="/"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      <h1
                                        className="title"
                                      >
                                        IoT
                                      </h1>
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                              <div
                                className="navbar-burger burger"
                                data-target="navbar-menu"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <span />
                                <span />
                                <span />
                              </div>
                            </div>
                            <div
                              className="navbar-menu"
                              id="navbar-menu"
                            >
                              <div
                                className="navbar-start"
                              >
                                <NavLink
                                  activeClassName="navbar-item-active"
                                  aria-current="page"
                                  className="navbar-item"
                                  to="/things"
                                >
                                  <Route
                                    path="\\\\/things"
                                  >
                                    <Link
                                      aria-current="page"
                                      className="navbar-item navbar-item-active"
                                      replace={false}
                                      style={Object {}}
                                      to="/things"
                                    >
                                      <a
                                        aria-current="page"
                                        className="navbar-item navbar-item-active"
                                        href="/things"
                                        onClick={[Function]}
                                        style={Object {}}
                                      >
                                        Things
                                      </a>
                                    </Link>
                                  </Route>
                                </NavLink>
                                <NavLink
                                  activeClassName="navbar-item-active"
                                  aria-current="page"
                                  className="navbar-item"
                                  to="/stats"
                                >
                                  <Route
                                    path="\\\\/stats"
                                  >
                                    <Link
                                      aria-current={null}
                                      className="navbar-item"
                                      replace={false}
                                      to="/stats"
                                    >
                                      <a
                                        aria-current={null}
                                        className="navbar-item"
                                        href="/stats"
                                        onClick={[Function]}
                                      >
                                        Stats
                                      </a>
                                    </Link>
                                  </Route>
                                </NavLink>
                              </div>
                              <div
                                className="navbar-end"
                              >
                                <div
                                  className="navbar-item"
                                >
                                  <div
                                    className="field"
                                  >
                                    <p
                                      className="control"
                                    >
                                      <button
                                        className="button is-primary"
                                        id="logout-button"
                                        onClick={[Function]}
                                      >
                                        <span
                                          className="icon"
                                        >
                                          <i
                                            className="fa fa-user"
                                          />
                                        </span>
                                        <span>
                                          Logout
                                        </span>
                                      </button>
                                    </p>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </nav>
                        </Navbar>
                      </Connect(Navbar)>
                    </Route>
                  </withRouter(Connect(Navbar))>
                  <main
                    className="container"
                  >
                    <Switch>
                      <Route
                        component={[Function]}
                        computedMatch={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/things",
                            "url": "/things",
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/things",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        path="/things"
                      >
                        <Things
                          history={
                            Object {
                              "action": "REPLACE",
                              "block": [Function],
                              "canGo": [Function],
                              "createHref": [Function],
                              "entries": Array [
                                Object {
                                  "hash": "",
                                  "pathname": "/things",
                                  "search": "",
                                  "state": undefined,
                                },
                              ],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "index": 0,
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/things",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "/things",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": true,
                              "params": Object {},
                              "path": "/things",
                              "url": "/things",
                            }
                          }
                        />
                      </Route>
                    </Switch>
                  </main>
                </div>
              </Main>
            </Provider>
          </App>
        </Connect(App)>
      </Route>
    </withRouter(Connect(App))>
  </Router>
</MemoryRouter>
`;

exports[`containers/app renders the app without auth 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Connect(App))
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Route>
        <Connect(App)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <App
            dispatch={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isAuth={false}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            store={
              Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              }
            }
            storeSubscription={
              Subscription {
                "listeners": Object {
                  "clear": [Function],
                  "get": [Function],
                  "notify": [Function],
                  "subscribe": [Function],
                },
                "onStateChange": [Function],
                "parentSub": undefined,
                "store": Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                },
                "unsubscribe": [Function],
              }
            }
          >
            <Provider
              store={
                Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
              }
            >
              <Connect(Login)>
                <Login
                  isAuth={[Function]}
                  isLoading={false}
                  login={[Function]}
                  logout={[Function]}
                  password={null}
                  setPassword={[Function]}
                  setShowError={[Function]}
                  setUsername={[Function]}
                  shouldShowError={false}
                  username={null}
                >
                  <section
                    className="hero is-fullheight"
                  >
                    <div
                      className="hero-body"
                    >
                      <div
                        className="container has-text-centered"
                      >
                        <div
                          className="column is-4 is-offset-4"
                        >
                          <div
                            className="box"
                          >
                            <p
                              className="title is-1"
                            >
                              IoT
                            </p>
                            <form>
                              <div
                                className="field"
                              >
                                <div
                                  className="control has-icons-left"
                                >
                                  <input
                                    className="input is-large"
                                    id="username-input"
                                    onChange={[Function]}
                                    placeholder="Username"
                                    value=""
                                  />
                                  <span
                                    className="icon is-large is-left"
                                  >
                                    <i
                                      className="fas fa-user"
                                    />
                                  </span>
                                </div>
                              </div>
                              <div
                                className="field"
                              >
                                <div
                                  className="control has-icons-left"
                                >
                                  <input
                                    className="input is-large"
                                    id="password-input"
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                                    value=""
                                  />
                                  <span
                                    className="icon is-large is-left"
                                  >
                                    <i
                                      className="fas fa-lock"
                                    />
                                  </span>
                                </div>
                              </div>
                              <button
                                className="button is-block is-primary is-large is-fullwidth"
                                disabled={false}
                                id="login-button"
                                onClick={[Function]}
                              >
                                Login
                              </button>
                            </form>
                          </div>
                        </div>
                      </div>
                    </div>
                    <Modal
                      isActive={false}
                      messageStyle={
                        Object {
                          "is-danger": true,
                        }
                      }
                      onCloseClick={[Function]}
                      subTitle="Invalid Credentials"
                      title="Error"
                    >
                      <div
                        className="modal"
                      >
                        <div
                          className="modal-background"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          role="button"
                          tabIndex={0}
                        />
                        <div
                          className="modal-content"
                        >
                          <article
                            className="message is-medium is-danger"
                          >
                            <div
                              className="message-header"
                            >
                              <p>
                                Error
                              </p>
                            </div>
                            <div
                              className="message-body has-text-centered"
                            >
                              Invalid Credentials
                            </div>
                          </article>
                        </div>
                        <button
                          aria-label="close"
                          className="modal-close is-large"
                          onClick={[Function]}
                        />
                      </div>
                    </Modal>
                  </section>
                </Login>
              </Connect(Login)>
            </Provider>
          </App>
        </Connect(App)>
      </Route>
    </withRouter(Connect(App))>
  </Router>
</MemoryRouter>
`;
